// react-hamburger/lib/components.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TopContentContainer = exports.TopContainer = exports.TopBar = exports.LinkContainerVertical = exports.LinkContainerHorizontal = exports.LinkContainer = exports.HamburgerIcon = exports.Bar = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styles = require("./styles");

var Bar = _styledComponents.default.div.withConfig({
  displayName: "components__Bar",
  componentId: "sc-1hsxvt7-0"
})(["width:100%;", ";"], function (_ref) {
  var color = _ref.color,
      height = _ref.height,
      radius = _ref.radius,
      bars = _ref.theme.bars;
  return "\n    background: ".concat(color || bars.color, ";\n    border-radius: ").concat(radius || bars.radius, "px;\n    height: ").concat(height || bars.height, "px;\n  ");
});

exports.Bar = Bar;

var HamburgerIcon = _styledComponents.default.div.withConfig({
  displayName: "components__HamburgerIcon",
  componentId: "sc-1hsxvt7-1"
})(["cursor:pointer;display:flex;flex-direction:column;justify-content:space-around;z-index:100;", ";", ""], function (_ref2) {
  var height = _ref2.height,
      dimensions = _ref2.theme.hamburger.dimensions,
      width = _ref2.width;
  return "\n    height: ".concat(height || dimensions, "px;\n    width: ").concat(width || dimensions, "px;\n  ");
}, function (_ref3) {
  var maxWidth = _ref3.maxWidth,
      right = _ref3.right,
      slide = _ref3.slide,
      _ref3$theme = _ref3.theme,
      hamburger = _ref3$theme.hamburger,
      linkContainer = _ref3$theme.linkContainer;
  return (// eslint-disable-next-line no-nested-ternary
    slide || hamburger.slide ? right ? (0, _styles.horizontalSlideTransition)(-maxWidth || -linkContainer.maxWidth, 0, 'px') : (0, _styles.horizontalSlideTransition)(0, maxWidth || linkContainer.maxWidth, 'px') : undefined
  );
});

exports.HamburgerIcon = HamburgerIcon;

var LinkContainer = _styledComponents.default.div.withConfig({
  displayName: "components__LinkContainer",
  componentId: "sc-1hsxvt7-2"
})(["position:fixed;z-index:98;", ";", ";"], function (_ref4) {
  var color = _ref4.color,
      padding = _ref4.padding,
      linkContainer = _ref4.theme.linkContainer;
  return "\n    background: ".concat(color || linkContainer.color, ";\n    padding: ").concat(padding || linkContainer.padding, "px;\n  ");
}, _styles.linkContainerSpacing);

exports.LinkContainer = LinkContainer;
var LinkContainerHorizontal = (0, _styledComponents.default)(LinkContainer).withConfig({
  displayName: "components__LinkContainerHorizontal",
  componentId: "sc-1hsxvt7-3"
})(["", ";", ""], function (_ref5) {
  var maxWidth = _ref5.maxWidth,
      right = _ref5.right,
      _ref5$theme = _ref5.theme,
      linkContainer = _ref5$theme.linkContainer,
      hamburger = _ref5$theme.hamburger,
      width = _ref5.width;
  return "\n    height: 100vh;\n    max-width: ".concat(maxWidth || linkContainer.maxWidth, "px;\n    right: ").concat(right || hamburger.location === 'right' ? "".concat(-linkContainer.maxWidth, "px") : '', ";\n    width: ").concat(width || linkContainer.width, "%;\n  ");
}, (0, _styles.horizontalSlideTransition)(-100, 0, '%'));
exports.LinkContainerHorizontal = LinkContainerHorizontal;
var LinkContainerVertical = (0, _styledComponents.default)(LinkContainer).withConfig({
  displayName: "components__LinkContainerVertical",
  componentId: "sc-1hsxvt7-4"
})(["width:100%;", ";"], (0, _styles.verticalSlideTransition)(-100, 0, '%'));
exports.LinkContainerVertical = LinkContainerVertical;

var TopBar = _styledComponents.default.div.withConfig({
  displayName: "components__TopBar",
  componentId: "sc-1hsxvt7-5"
})(["align-items:center;position:relative;", ";", ""], function (_ref6) {
  var color = _ref6.color,
      height = _ref6.height,
      gutter = _ref6.gutter,
      topBar = _ref6.theme.topBar;
  return "\n    background: ".concat(color || topBar.color, ";\n    height: ").concat(height || topBar.height, "px;\n    padding-left: ").concat(gutter || topBar.gutter, "px;\n    padding-right: ").concat(gutter || topBar.gutter, "px;\n  ");
}, _styles.topBarPositioning);

exports.TopBar = TopBar;

var TopContainer = _styledComponents.default.div.withConfig({
  displayName: "components__TopContainer",
  componentId: "sc-1hsxvt7-6"
})(["", ""], _styles.topBarPositioning);

exports.TopContainer = TopContainer;

var TopContentContainer = _styledComponents.default.div.withConfig({
  displayName: "components__TopContentContainer",
  componentId: "sc-1hsxvt7-7"
})(["flex:1;"]);

exports.TopContentContainer = TopContentContainer;
